packages/slate/src/changes/with-schema.js

```diff
@@ -67,42 +66,26 @@ function normalizeNodeAndChildren(change, node, schema) {
     normalizeNode(change, node, schema)
     return
   }
+  const checkedKeys = []
+  let child = node.nodes.first()
+
+  let nodePath = change.value.document.getPath(node.key)
 
   // We can't just loop the children and normalize them, because in the process
   // of normalizing one child, we might end up creating another. Instead, we
   // have to normalize one at a time, and check for new children along the way.
   // PERF: use a mutable array here instead of an immutable stack.
-  const keys = node.nodes.toArray().map(n => n.key)
-
-  // While there is still a child key that hasn't been normalized yet...
-  while (keys.length) {
-    const { size } = change.operations
-    let key
-
-    // PERF: use a mutable set here since we'll be add to it a lot.
-    let set = new Set().asMutable()
-
-    // Unwind the stack, normalizing every child and adding it to the set.
-    while (key = keys[0]) {
-      const child = node.getChild(key)
-      normalizeNodeAndChildren(change, child, schema)
-      set.add(key)
-      keys.shift()
-    }
-
-    // Turn the set immutable to be able to compare against it.
-    set = set.asImmutable()
-
-    // PERF: Only re-find the node and re-normalize any new children if
-    // operations occured that might have changed it.
-    if (change.operations.size > size) {
-      node = refindNode(change, node)
-
-      // Add any new children back onto the stack.
-      node.nodes.forEach((n) => {
-        if (set.has(n.key)) return
-        keys.unshift(n.key)
-      })
+  while (node && child) {
+    const lastSize = change.operations.size
+    normalizeNodeAndChildren(change, child, schema)
+    checkedKeys.push(child.key)
+    if (lastSize === change.operations.size) {
+      child = node.nodes.get(node.nodes.indexOf(child) + 1)
+    } else {
+      const cacheNodeAndNodePath = refindNode(change, node, nodePath)
```

Comment:
Can you change this variable to `result`?

Reply:
Yes, I will.